app/authorization/server/startup.js

```diff
@@ -155,27 +153,34 @@ Meteor.startup(function() {
 	const createSettingPermission = function(setting, previousSettingPermissions) {
 		const permissionId = getSettingPermissionId(setting._id);
 		const permission = {
-			_id: permissionId,
 			level: CONSTANTS.SETTINGS_LEVEL,
 			// copy those setting-properties which are needed to properly publish the setting-based permissions
 			settingId: setting._id,
 			group: setting.group,
 			section: setting.section,
 			sorter: setting.sorter,
+			roles: [],
 		};
 		// copy previously assigned roles if available
 		if (previousSettingPermissions[permissionId] && previousSettingPermissions[permissionId].roles) {
 			permission.roles = previousSettingPermissions[permissionId].roles;
-		} else {
-			permission.roles = [];
 		}
 		if (setting.group) {
 			permission.groupPermissionId = getSettingPermissionId(setting.group);
 		}
 		if (setting.section) {
 			permission.sectionPermissionId = getSettingPermissionId(setting.section);
 		}
-		Permissions.upsert(permission._id, { $set: permission });
+
+		const existent = Permissions.findOne({
+			_id: permissionId,
+			...permission,
+		}, { fields: { _id: 1 } });
+
+		if (!existent) {
+			Permissions.upsert({ _id: permissionId }, { $set: permission });
```

Comment:
insert?

Reply:
The permission may exist with different properties